src/clj/fluree/db/json_ld/migrate/sid.cljc

```diff
@@ -115,11 +118,13 @@
           branch            (or (keyword (get-first-value first-commit const/iri-branch))
                                 :main)
           ledger            (<? (jld-ledger/->ledger conn ledger-alias {:branch branch}))
-          indexer           (:indexer ledger)
-          db                (<? (merge-commits ledger all-commit-tuples))
           commit-opts*      (assoc commit-opts :branch branch)
-          update-commit     (commit/update-commit-fn db commit-opts*)
-          indexed-db        (<? (indexer/do-index indexer db {:changes-ch    changes-ch
-                                                              :update-commit update-commit}))]
-      (ledger/-db-update ledger indexed-db)
+          tuples-chans      (->> all-commit-tuples
+                                 (map (fn [commit-tuple]
+                                        [commit-tuple (async/chan)])))
+          changes-chs       (map second tuples-chans)
+          _                 (-> (async/merge changes-chs)
+                                (async/pipe changes-ch))
```

Comment:
Might be better as:

```clojure
(-> changes-chs
    async/merge
    (async/pipe changes-ch))
```

Or:

```clojure
(let [merged (async/merge changes-chs)
      _      (async/pipe merged changes-ch)
```

Or even just:

```clojure
(async/pipe (async/merge changes-chs) changes-ch)
```

Reply:
Went with your first option